serverAkkaHttp/src/main/scala/RPCRouter.scala

```diff
@@ -16,7 +16,21 @@ import scala.concurrent.Future
 
 import io.circe.Json
 
-object RouteGenerators {
+trait Router extends RPCServer with PathMacro with MetaDataMacro {
+  def tp: Seq[String]
+  def methodsMetaData: Map[String, MethodMetaData]
+  def routes: autowire.Core.Router[Json]
+  def path: String = tp.last
+
+  def buildRoute: Route = handleExceptions(exceptionHandler) {
+    pathPrefix(path) {
+      methodsMetaData map {
```

Comment:
@gabro 
I'm aware of the problem.
No, there's no guarantee `methodsMetaData ` is empty... though you need to pass an empty controller to `deriveRoute` to have it empty. That's unsound.

I don't think `reduceOption` is the solution (Once I have an option I wouldn't know what to do other than throwing an exception). I've been searching for a default `Route.empty` but I couldn't found it (to be honest I didn't spent much time searching for it :P).